lib/utilities/async_button.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:flutter/material.dart';
+
+class AsyncButton extends StatefulWidget {
+  final void Function() onPressedFunction;
+  final String label;
+
+  const AsyncButton({required this.onPressedFunction, required this.label, Key? key})
```

Comment:
🔧 We need to be able to disable the button from the outside, e.g. if the form is invalid. This was already done on the open channel button, e.g. if the channel amount was not set.